### ISSUE TITLE
Feature request: close modal dialog by clicking outside

### ISSUE THREAD

**Issue by kam1sh** [rjaros/kvision#165]
Hi! Me again.

I'm still playing with your examples repo and recently I found out that `pl.treksoft.kvision.modal.Dialog` cannot be closed by just clicking outside.

I've tried to implement this by myself (via jQuery bindings), but failed, because I don't know enough jQuery =/

Can you implement this by additional flag in `Dialog` or `Modal` classes? Or maybe it's better to add event handler with something like `clickOutside`? I mean this:
```kotlin
onEvent {
    clickOutside = {
        hide()
    }
}
```

Thank you.

**Comment by rjaros** [comment]
You should be able to close `Modal` or `Dialog` when clicking outside. It's controlled by the `escape` parameter, which is `true` by default. You can test this in the Showcase example:
https://rjaros.github.io/kvision-examples/showcase/#!/windows

If it's not working for you, could you please provide more details or some code?

**Comment by kam1sh** [comment]
Take a addressbook-fullstack-ktor as an example, file `src/frontendMain/kotlin/com/example/Security.kt`, and replace in `LoginWindow`  `closeButton` and `escape` with `true`. For me it doesn't work, I can't close login window by clicking outside, only by close icon. But maybe problem is in my platform (Vivaldi browser on Linux)?..

**Comment by rjaros** [comment]
The `LoginWindow` is connected to the `SecurityMgr`:

https://github.com/rjaros/kvision-examples/blob/6653ff4d4e0678277312ba63c71a1ee001a6902d/addressbook-fullstack-ktor/src/frontendMain/kotlin/com/example/Security.kt#L129

And the security manager runs this loop (coroutines based so without blocking):

https://github.com/rjaros/kvision/blob/f5d791e50d16e38c0135909f30a3199cd7826a54/kvision-modules/kvision-common-remote/src/jsMain/kotlin/pl/treksoft/kvision/remote/Security.kt#L83

until the user is authenticated. So generally the login window is closed when you click outside, but it's immediately opened again. There is slight difference in event handling between clicking outside and clicking close button, that's why you can't notice any action when clicking outside (but you can see fast changes of the html with chrome dev tools panel).

**Comment by kam1sh** [comment]
Hmm... In my project I don't use `SecurityMgr`.

Here's another minimal example, based on template: https://github.com/kam1sh/kvision-sandbox/tree/login-form
```kotlin
@Serializable
data class Credentials(val username: String, val password: String)

class LoginWindow : Dialog<Credentials>() {
    private val loginPanel: FormPanel<Credentials>
    private val loginButton: Button

    init {
        loginPanel = formPanel {
            add(Credentials::username, Text(label = "Login:"), required = true)
            add(Credentials::password, Password(label = "Password:"))
        }
        loginButton = Button("Login", "fas fa-check", ButtonStyle.PRIMARY).onClick {
            if (loginPanel.validate()) {
                println(loginPanel.getData())
            }
        }
        addButton(loginButton)
    }
}

class App : Application() {
    init {
        require("css/kvapp.css")
    }

    override fun start() {
        I18n.manager =
            DefaultI18nManager(
                mapOf(
                    "pl" to require("i18n/messages-pl.json"),
                    "en" to require("i18n/messages-en.json")
                )
            )

        root("kvapp") {
            LoginWindow().show()
        }
    }
}

```

Login window without SecurityMgr, but I still can't close it.

**Comment by rjaros** [comment]
You have found a bug in the `Dialog` class. The order of constructor parameters `scrollable`, `centered` and `escape` is mixed up. So at the moment you can create your `Dialog` with `scrollable = true` and it will work ;-)
Fix is on the way.

**Comment by rjaros** [comment]
Released with 3.11.0
